### PR TITLE
Fix Strict Error isTempFile

### LINKED ISSUE #1164: Strict Error TemporaryFileFilterPlugin
preg_match() expects parameter 2 to be string, null given

### PR DESCRIPTION
**Pull request by WorksDev**
- fix #1164

### CODE REVIEW

File: lib/DAV/TemporaryFileFilterPlugin.php [hunk 1]
```diff
@@ -173,7 +173,7 @@ protected function isTempFile($path)
         list(, $tempPath) = Uri\split($path);
 
         foreach ($this->temporaryFilePatterns as $tempFile) {
-            if (preg_match($tempFile, $tempPath)) {
+            if (preg_match($tempFile, $tempPath ?? '')) {
```
Comment (staabm) [inline lib/DAV/TemporaryFileFilterPlugin.php]: this case should be handled above the loop and we should early return then

**Comment by staabm** [comment]
thx for working on a fix. please find my comments in the PR.

File: lib/DAV/TemporaryFileFilterPlugin.php (hunk 1 above)
Comment (WorksDev) [reply lib/DAV/TemporaryFileFilterPlugin.php]: You're right about that. Changed. Thank you.

File: tests/Sabre/DAV/TemporaryFileFilterTest.php [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 namespace Sabre\DAV;
 
+use Sabre\DAV\Browser\Plugin;
```
Comment (staabm) [inline tests/Sabre/DAV/TemporaryFileFilterTest.php]:
```suggestion
use Sabre\DAV\Browser;
```

File: tests/Sabre/DAV/TemporaryFileFilterTest.php [hunk 3]
```diff
@@ -99,6 +100,17 @@ public function testPutGet()
         $this->assertEquals('Testing new file', stream_get_contents($this->response->body));
     }
 
+    public function testGetWithBrowserPlugin()
+    {
+        $this->server->addPlugin(new Plugin());
```
Comment (staabm) [inline tests/Sabre/DAV/TemporaryFileFilterTest.php]:
```suggestion
        $this->server->addPlugin(new Browser\Plugin());
```

**Comment by staabm** [comment]
looks good, waiting for a green build. thx!

**Comment by staabm** [comment]
awesome, thx

**Comment by WorksDev** [comment]
@staabm is there already a planned date for a new release of the latest fixes? Thanks for the info in advance.

**Comment by staabm** [comment]
lets see https://github.com/sabre-io/dav/issues/1177

**Pull request merged**